engines/dragons/screen.cpp

```diff
@@ -0,0 +1,411 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+#include "common/endian.h"
+#include "common/system.h"
+#include "common/rect.h"
+#include "engines/util.h"
+#include "dragons/dragons.h"
+#include "dragons/scene.h"
+#include "dragons/screen.h"
+
+namespace Dragons {
+
+Screen::Screen() {
+	_pixelFormat = Graphics::PixelFormat(2, 5, 5, 5, 1, 10, 5, 0, 15);
+	initGraphics(320, 200, &_pixelFormat);
```

Comment:
You will probably want to use a more common pixel format for the screen surface. Some backends may not recognize this one and refuse to start the game. Some others will have to to color conversion in software, which is likely to cause slowness on low performance devices.

Reply:
Good point, any recommendation on which one to use? I picked this because it aligns with the PS1 format used by the original game engine.

Reply:
RGB565 is a pretty good choice. It's required to be supported by spec in all versions of OpenGL (including ES) so most devices have it.

Reply:
Hmm I can convert to rgb565 but it will require a bit of rework. I'm using the palette in rgb555+A in a few places. I wonder if just removing the alpha from this format would help? I'm not actually using the alpha bit here anyway.

Reply:
Yes, using RGB555 (the same you use without the alpha bit) as the screen surface would work on most platforms. It's just more likely to trigger software color conversion in the backend than RGB565.